### PR TITLE
Fix mod select overlay overflowing toolbar at max ui scale

### LINKED ISSUE #7369: Mod select overlay overflows at max ui scale
Just want to make sure this is right. First, we can make it relative size (not 100% though). We can then share the resize implementation in `ChatOverlay`, so portrait players won't be stuck with the higher height.

**Screenshots or videos showing encountered issue:** 
osu_2019-12-26_16-55-20

### PR DESCRIPTION
**Pull request by Joehuu**
Fixes https://github.com/ppy/osu/issues/7369. Uses grid container instead of what I proposed (making it resizeable). May be good to do in the future, when more mods are introduced.

For the triangles, all other uses which resize (direct, etc.) have relative axes on both. From what I know, the densities of those are wrong when opened too.

tI9Oacf3ai

### CODE REVIEW

**Review by bdach — DISMISSED** [review]
Unfortunately this doesn't seem to be working properly, which you've even demonstrated in the PR description. When increasing UI scale mods can overflow the screen width and be partially or even fully off screen ([see video](https://streamable.com/moxnm)).

**Comment by peppy** [comment]
Seems that some mods and text in the dialog end up off screen now?

**Comment by Joehuu** [comment]
Not sure how you missed the toolbar in the PR title, but this just fixes the height of the overlay to not exceed the toolbar height. The issue you mentioned already exists in master and may just be another `FillFlowContainer` problem.

**Comment by bdach** [comment]
Are you sure? You had submitted #6700 before, which seemed to be a fix for that same issue - doesn't that mean that this regressed somewhere?

**Comment by bdach** [comment]
Upon closer inspection this seems to always have been broken. The flow brokenness will most likely end up being another case of ppy/osu-framework#223; sorry for the confusion.

File: osu.Game/Screens/Select/SongSelect.cs [hunk 1]
```diff
@@ -221,23 +221,37 @@ private void load(AudioManager audio, DialogOverlay dialog, OsuColour colours, S
 
             if (ShowFooter)
             {
-                AddRangeInternal(new[]
+                AddRangeInternal(new Drawable[]
                 {
-                    FooterPanels = new Container
+                    new GridContainer // used for max width implementation
```
Comment (peppy) [inline osu.Game/Screens/Select/SongSelect.cs]: max height?

File: osu.Game/Screens/Select/SongSelect.cs [hunk 2]
```diff
@@ -221,23 +221,37 @@ private void load(AudioManager audio, DialogOverlay dialog, OsuColour colours, S
...
-                        AutoSizeAxes = Axes.Y,
-                        Margin = new MarginPadding { Bottom = Footer.HEIGHT },
-                        Children = new Drawable[]
+                        RelativeSizeAxes = Axes.Both,
+                        RowDimensions = new[]
                         {
-                            BeatmapOptions = new BeatmapOptionsOverlay(),
-                            ModSelect = new ModSelectOverlay
+                            new Dimension(),
+                            new Dimension(GridSizeMode.Relative, 1f, maxSize: 560),
```
Comment (peppy) [inline osu.Game/Screens/Select/SongSelect.cs]: need to know where 560 comes from, via a shared constant or otherwise.

**Review by peppy — DISMISSED** [review]

**Review by smoogipoo — APPROVED** [review]
Seems to look okay 👍

**Pull request merged**